test/unittests/execute_test.cpp

```diff
@@ -320,6 +320,346 @@ TEST(execute, i64_load_trap)
     ASSERT_TRUE(trap);
 }
 
+TEST(execute, i32_load8_s)
+{
+    Module module;
+    module.memorysec.emplace_back(Memory{{1, 1}});
+    module.codesec.emplace_back(
+        Code{0, {Instr::local_get, Instr::i32_load8_s, Instr::end}, {0, 0, 0, 0, 0, 0, 0, 0}});
+
+    auto instance = instantiate(module);
+    instance.memory[0] = 0x80;
+    instance.memory[1] = 0xf1;
+    const auto [trap, ret] = execute(instance, 0, {0});
+
+    ASSERT_FALSE(trap);
+    ASSERT_EQ(ret.size(), 1);
+    ASSERT_EQ(static_cast<int32_t>(ret[0]), -128);
+}
+
+TEST(execute, i32_load8_s_trap)
+{
+    Module module;
+    module.memorysec.emplace_back(Memory{{1, 1}});
+    module.codesec.emplace_back(
+        Code{0, {Instr::local_get, Instr::i32_load8_s, Instr::end}, {0, 0, 0, 0, 0, 0, 0, 0}});
+
+    auto instance = instantiate(module);
+    instance.memory[0] = 0x80;
+    instance.memory[1] = 0xf1;
+    const auto [trap, ret] = execute(instance, 0, {65537});
```

Comment:
I suggest to merge each pair of non-trap and trap test into one, as they differ only by passed argument